universum/modules/vcs/swarm.py

```diff
@@ -175,17 +175,21 @@ def code_report_to_review(self, report):
     def report_result(self, result, report_text=None, no_vote=False):
         # Opening links, sent by Swarm
         # Does not require login to Swarm; changes "Automated Tests" icon
-        if result:
-            link = self.settings.pass_link
-        else:
-            link = self.settings.fail_link
-
-        if link is not None:
-            self.out.log("Swarm will be informed about build status by URL " + link)
-            utils.make_request(link, critical=False)
+        # Should not be applied to non-latest revision
+        if self.is_latest_version():
+            if result:
+                link = self.settings.pass_link
+            else:
+                link = self.settings.fail_link
+
+            if link is not None:
+                self.out.log("Swarm will be informed about build status by URL " + link)
+                utils.make_request(link, critical=False)
+            else:
+                self.out.log("Swarm will not be informed about build status because " +
+                             "the '{0}' link was not provided".format("PASS" if result else "FAIL"))
         else:
-            self.out.log("Swarm will not be informed about build status because " +
-                         "the '{0}' link was not provided".format("PASS" if result else "FAIL"))
+            self.out.log("Review test status will not be changed for review revision is not latest")
```

Comment:
This message is printed even if user did not supply PASS or FAIL parameter.
This seems counterintuitive.

Reply:
I think this is more intuitive, because if the review revision is not latest, user doesn't have to provide such links, and therefore shouldn't get that warning.

Reply:
Previous behavior: if user doesn't provide the link, print message that the link is not provided regardless of the review version.
New behavior: If the review is latest, do as before, but if it is not latest - totally ignore links.
Frankly, either type of the new behavior (with links absence having higher priority over not latest review version or the opposite) is fine on its own. But considering we already have some behavior, to me it looks better to give the review version lower priority.